packages/ui/src/common/providers/modal/provider.tsx

```diff
@@ -24,8 +25,16 @@ export const ModalContextProvider = (props: Props) => {
         setModalData(modalCall.data)
       }
     },
-    hideModal: () => {
-      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames)) {
+    hideModal: (showConfirmOrAny = false) => {
+      const preferShowConfirm = typeof showConfirmOrAny === 'boolean' ? showConfirmOrAny : false
+      const [state] = currentModalMachine ?? []
+      if (
+        !preferShowConfirm ||
+        isClosing ||
+        !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) ||
+        state?.matches('success') ||
+        state?.matches('beforeTransaction')
```

Comment:
I think we want to keep the validation here too. The goal is to avoid closing the modal by mistake and right before the transaction is the worse moment for it to happen.

Reply:
Yes